### DISCUSSION TITLE
Display `MOWNR` prints as "0WNR" - HERO 11

### DISCUSSION THREAD

**Discussion by sT0v** [gopro/labs#1185] [Q&A] [answered]
It is so cool to see that GoPro Labs enables users to encode logic via QR codes. I stumbled across this great feature when I wanted to mute livestreams and found `DAUD=1`.

To familiarize myself with the language, I'd like to display/repeat the `MOWNR` message for 5 secs.

As a test, I wrote this:

1. `=Tt:Q=T<5"$T"!R` (does not compile?)
2. `=Tt:Q"$T"T<5!R` (prints T once but condition doesn't evaluate or repeat)
3. `=Tt:Q<T5"$T"!R` (GOOD - but hangs once condition is met "Labs Processing") How can this exit?
4. `=Tt:Q"$T"<T5!R!X` (GREAT, now the only problem is the OWNR str interpolation)
I was expecting that example (1) would print `T` until the `T<5` condition failed (user error probably). For testing I'm using `t-Q`, but in prod it would be `t-B` for time since boot.

Assuming that worked, I should 
then just do `"$MOWNR"` in place of `"$T"`.

Partial solution:
Use `!5E` instead of conditional repeat.
```
=T:t-Q"$T"!5E
>>> 0 for 5 sec (good)

"$OWNR"!5E 
>>> 0WNR for 5 sec (bad)
```
The problem remains that `OWNR` variable isn't interpolated correctly into the string.

Even though, `!5E` is probably the best option for this example. I'd like to know how the conditional repeat could work.

Finally, I'd have amended the script to do this unless `FAST` was enabled. This is least important and can be ignored.

Thanks for your incredible work. Any feedback would be greatly appreciated.

Thank you.

**Comment by dnewman-gpsw** [comment, marked as answer]
1. =Tt:Q=T<5"$T"!R (does not compile?)

It is an interpreted language, so no compile.

This is the corrected code: =Tt:Q<T5"$T"!R  
```
=Tt:Q  //assigned(=) T with t:Q
<T5    //conditional (<) T<5
"$T"   //print "$T"
!R     // repeat
```

2. =Tt:Q"$T"T<5!R

The corrected code:  =Tt:Q"$T"<T5!R

Version 2 is the better code, as it will stop looping after five seconds, where 1. loop forever, but only prints for 5 seconds.

So you just writing your conditionals incorrectly. There is a reason for the reversed notation, as it makes interpreter simpler to parse and delimit the code. All code segments start with either ! (action), <,>,== (condition), $A= or $ABCD= (assignment), "" print, or a lower case character is a setting. Then all numbers, :, or uppercase characters are part of the previous code segment.

MOWNR is not the fourCC, OWNR is.

Printing directly form a FourCC is not currently support (although that could be added).  Only supported variable printing is this form:
=B$BITR"bitrate $B"

The above works from numbers, but not for strings.  Variables are only stored as floats.

Your example of "$OWNR" does seem a reasonable use case, so I will see if that can be supported in a future release.

**Reply by sT0v** [reply]
Thank you for your explanations. 
So if fourCC worked with strings the options would be:

```
# loop until T>5sec then exit
Tt:Q"$OWNR"<T5!R!X
# wait 5sec
Tt:Q"$OWNR"<T5!5E
```

In the interim the could 
`DSPL=5` work?

**Reply by sT0v** [reply]
```
!MOWNR="Owner: First Last\nPlease call or email if found\n555-555-5555 || phone@home.com"$DSPL=5
```

`$DSPL` doesn't seem to be remembered on restart

**Reply by dnewman-gpsw** [reply]
$ABCD=x is a temporary command (forgotten on the next boot.)
*ABCD=x is a permanent command (required if you want to remember over a boot.)

Also !MOWNR is old style.  The website uses the old version for the HERO8 and 9 users out there.

*OWNR="your info" is for all GP2 cameras (and MAX.)

**Reply by sT0v** [reply]
Great, thank you that works. Now is there a way to bypass the below? 

```
Features activated:
DSPL=5
```

If I used `FAST` then the `DSPL=5` for `OWNR` gets bypassed.

`FEAT=0` does not achieve this.

Thanks for your prompt support. At this point, unless I hit a non-trivial hurdle, I'll leave my iterative development questions to a minimum. Thanks again. Will be on the lookout for the `SHEL` command!

**Reply by dnewman-gpsw** [reply]
No, the "Features activated:" displays anything you have changed (it is a supposed to be a reminder.)

This is a hack of a workaround with *DSPL=0.
*OWNR="your info"
*OWNS="your info"
*OWNT="your info"
*OWNU="your info"
*OWNV="your info"

It will display them in order (1 second each.)

You can used this to page through owner information.

so rather than: "Owner: First Last\nPlease call or email if found\n555-555-5555 || phone@home.com"

*OWNR="Owner: First Last"
*OWNS="Please call\n(telephone)"
*OWNT="or email if found\n(email)"
*OWNU="name\n(fax?)"
*OWNV="reward ($50)"

**Comment by sT0v** [comment]
TLDR

`=Tt:Q"$T"<T5!R!X` - Good

This works as expected per the logic. The issue now is the `OWNR` str interpolation displays as "0WNR"

`=Tt:Q"$MOWNR"<T5!R!X` - Buggy?

**Reply by dnewman-gpsw** [reply]
You needed to read, as I explained that "$MOWNR" is wrong, and "$OWNR" is not supported in this firmware, bit it will be in the next, just got that working.
